runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/SamzaMetricsBundleProgressHandler.java

```diff
@@ -90,7 +90,9 @@ public void onProgress(BeamFnApi.ProcessBundleProgressResponse progress) {}
   public void onCompleted(BeamFnApi.ProcessBundleResponse response) {
     response.getMonitoringInfosList().stream()
         .filter(monitoringInfo -> !monitoringInfo.getPayload().isEmpty())
-        .forEach(this::parseAndUpdateMetric);
+        .map(this::parseAndUpdateMetric)
+        .distinct()
+        .forEach(samzaMetricsContainer::updateMetrics);
```

Comment:
why do we need to update the metrics again here? IIRC, beam native counter, gauge and distribution metric is integrated with samza metrics already

Reply:
Metrics doesn't work without this line in local testing.

samzaMetricsContainer probably has copies of both a samza metrics type and a beam metrics type. This line triggers the copy from beam to samza metrics